src/test/java/seedu/address/model/room/RoomNumberOrTagsContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,123 @@
+        assertTrue(predicate.test(new RoomBuilder().withRoomNumber("01-021").build()));
+
+        // Only one matching keyword
+        predicate = new RoomNumberOrTagsContainsKeywordsPredicate(Arrays.asList("01", "02"));
+        assertTrue(predicate.test(new RoomBuilder().withRoomNumber("02-123").build()));
+
+    }
+
+    @Test
+    public void test_roomNumberDoesNotContainKeywords_returnsFalse() {
+        // Zero keywords
+        RoomNumberOrTagsContainsKeywordsPredicate predicate =
+                new RoomNumberOrTagsContainsKeywordsPredicate(Collections.emptyList());
+        assertFalse(predicate.test(new RoomBuilder().withRoomNumber("01-123").build()));
+
+        // Non-matching keyword
+        predicate = new RoomNumberOrTagsContainsKeywordsPredicate(Arrays.asList("01-123"));
+        assertFalse(predicate.test(new RoomBuilder().withRoomNumber("02-234").build()));
+
+        // Keywords match occupancy and room type, but does not match room number
+        predicate = new RoomNumberOrTagsContainsKeywordsPredicate(Arrays.asList("suite_ac", "n"));
+        assertFalse(predicate.test(new RoomBuilder()
+                .withRoomNumber("03-345")
+                .withRoomType("suite_ac")
+                .withOccupancyStatus("N")
+                .build()));
+    }
+
+    @Test
+    public void test_tagContainsKeywords_returnsTrue() {
+        // One keyword
+        RoomNumberOrTagsContainsKeywordsPredicate predicate =
+                new RoomNumberOrTagsContainsKeywordsPredicate(Collections.singletonList("tagOne"));
+        assertTrue(predicate.test(new RoomBuilder().withTags("tagOne").build()));
+
+        // Multiple keywords
+        predicate = new RoomNumberOrTagsContainsKeywordsPredicate(Arrays.asList("tagOne", "tagTwo"));
+        assertTrue(predicate.test(new RoomBuilder().withTags("tagOne").build()));
+
+        // Only one matching keyword
+        predicate = new RoomNumberOrTagsContainsKeywordsPredicate(Arrays.asList("tagOne", "tagTwo"));
+        assertTrue(predicate.test(new RoomBuilder().withTags("tagZero", "tagOne").build()));
+
+    }
+
+    @Test
+    public void test_tagDoesNotContainKeywords_returnsFalse() {
+        // Zero keywords
+        RoomNumberOrTagsContainsKeywordsPredicate predicate =
+                new RoomNumberOrTagsContainsKeywordsPredicate(Collections.emptyList());
+        assertFalse(predicate.test(new RoomBuilder().withTags("tagOne").build()));
+
+        // Non-matching keyword
+        predicate = new RoomNumberOrTagsContainsKeywordsPredicate(Arrays.asList("tagOne", "tagTwo"));
+        assertFalse(predicate.test(new RoomBuilder().withTags("tagThree").build()));
+
+        // Keywords match occupancy and room type, but does not match room number
+        predicate = new RoomNumberOrTagsContainsKeywordsPredicate(Arrays.asList("suite_ac", "n"));
+        assertFalse(predicate.test(new RoomBuilder()
+                .withRoomNumber("03-345")
```

Comment:
Probably better to use the static variables from the typical rooms class

Reply:
Will update.

Reply:
Updated.